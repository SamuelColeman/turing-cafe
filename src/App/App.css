@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../assets/cafe-img.jpeg');
  width: 100%;
  max-height: 100vh;
  overflow: scroll;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}