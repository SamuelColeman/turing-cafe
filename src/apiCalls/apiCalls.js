export const getReservations = () => {
	fetch('http://localhost:3001/api/v1/reservation')
		.then(response => response.json())
		.then(data => data)
}