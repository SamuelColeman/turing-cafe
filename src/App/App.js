import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form';
import { getReservations } from '../apiCalls/apiCalls'

export default class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [],
      isLoading: true,
      error: ''
    }
  }

  componentDidMount() {
    getReservations()
      .then(reservations => this.setState({ reservations, isLoading: false}))
      .catch(error => this.setState({ isLoading: false, error: error.message}))
  }

  render() {
    return (
      <main>
        <h1>Turing Cafe</h1>
        <Form />
      </main>
    )
  }
}
