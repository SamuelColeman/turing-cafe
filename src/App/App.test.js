import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
import { getReservations, postReservation } from '../apiCalls/apiCalls';

jest.mock('../apiCalls/apiCalls.js')

describe('App', () => {
	beforeEach(() => {
		getReservations.mockImplementation(() => {
			return Promise.resolve([{
				id: 1,
				name: "Christie",
				date: "12/29",
				time: "7:00",
				number: 12
				}])
		})
		const mockReservation = {
				id: 1,
				name: "Christie",
				date: "12/29",
				time: "7:00",
				number: 12
				}
	})

	it('should match the snapshot', () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	})

	it('should mount after retrieving', () => {
		shallow(<App />);
		expect(getReservations).toHaveBeenCalled();
	})

	it('should update state with new reservation when addReservation is invoked', async () => {
		const wrapper = shallow(<App />);
		const mockReservations = {
			id: 2,
			name: "Leta",
			date: "4/5",
			time: "7:00",
			number: 2
			}
		const expected = [{
				id: 1,
				name: "Christie",
				date: "12/29",
				time: "7:00",
				number: 12
				}, mockReservations]
		postReservation.mockImplementation(() => {
			return Promise.resolve(mockReservations)
		})
		await wrapper.instance().addReservation(mockReservations)
		expect(wrapper.state('reservations')).toEqual(expected)
	})
})