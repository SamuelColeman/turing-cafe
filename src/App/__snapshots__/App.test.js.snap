// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe
  </h1>
  <Form
    addReservation={[Function]}
  />
  <h1>
    Loading
  </h1>
  <Reservations
    reservations={Array []}
  />
</main>
`;
